test-ts/contracts/PrecompileCaller.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+// Calls GA CENNZ precompile
+contract PrecompileCaller {
+    // derived CENNZ token address on testnets (AssetId 16000)
+    // CcccccCc (prefix) + 00003e80 (assetId) + padding
+    // run through web3.utils.toChecksumAddress(..)
+    address cennz = 0xcCccccCc00003E80000000000000000000000000;
+
+    function balanceOfProxy(address who) public view returns (uint256) {
+        (bool success, bytes memory returnData) = cennz.staticcall(abi.encodeWithSignature("balanceOf(address)", who));
+        assembly {
+            if eq(success, 0) {
+                revert(add(returnData, 0x20), returndatasize())
```

Comment:
what does returndatasize() do?

Reply:
it returns the byte length of `returnData`